docs/core/examples/app3.go

```diff
@@ -0,0 +1,181 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	cmn "github.com/tendermint/tmlibs/common"
+	dbm "github.com/tendermint/tmlibs/db"
+	"github.com/tendermint/tmlibs/log"
+
+	bapp "github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+)
+
+const (
+	app3Name = "App3"
+)
+
+func NewApp3(logger log.Logger, db dbm.DB) *bapp.BaseApp {
+
+	// Create the codec with registered Msg types
+	cdc := NewCodec()
+
+	// Create the base application object.
+	app := bapp.NewBaseApp(app3Name, cdc, logger, db)
+
+	// Create a key for accessing the account store.
+	keyAccount := sdk.NewKVStoreKey("acc")
+	keyMain := sdk.NewKVStoreKey("main")
+	keyFees := sdk.NewKVStoreKey("fee")
+
+	// Set various mappers/keepers to interact easily with underlying stores
+	// TODO: Need to register Account interface or use different Codec
+	accountMapper := auth.NewAccountMapper(cdc, keyAccount, &auth.BaseAccount{})
+	accountKeeper := bank.NewKeeper(accountMapper)
+	metadataMapper := NewApp3MetaDataMapper(keyMain)
+	feeKeeper := auth.NewFeeCollectionKeeper(cdc, keyFees)
```

Comment:
The underlying implementation also performs serialization logic which makes it seem more of a mapper. Can Keepers also perform the same logic as Mappers? Seems like a bit of a blurry line.

Reply:
left as a TODO for now to clarify the feeKeeper (not sure if we can remove it from the base AnteHandler, or else we need to say something like we'll explain it after since we introduce the AnteHandler first